src/Chess.java

```diff
@@ -13,45 +33,112 @@ public static void play() {
         System.out.println();
 
         Scanner input = new Scanner(System.in);
-        Board chessBoard = new Board();
-        boolean checkmate = false;
-        boolean stalemate = false;
+        Boolean saved = false;
 
-        while (!checkmate && !stalemate) {
-            System.out.println(chessBoard);
+        while (!this.checkmate && !this.stalemate && !saved) {
+            System.out.println(this.chessBoard);
             System.out.println();
 
-            if (chessBoard.isInCheck(chessBoard.getCurrentPlayer())) {
-                System.out.println("" + chessBoard.getCurrentPlayer() + " is in check");
+            if (this.chessBoard.isInCheck(this.chessBoard.getCurrentPlayer())) {
+                System.out.println("" + this.chessBoard.getCurrentPlayer() + " is in check");
             }
 
             Piece piece = null;
 
-            System.out.print(chessBoard.getCurrentPlayer() + " to move: ");
-            Move move = Move.parse(input.next());
-            if (move != null) {
-                piece = chessBoard.getPiece(move.src);
+            System.out.print(this.chessBoard.getCurrentPlayer() + " to move: ");
+
+            String moveString = input.next();
+
+            if (moveString != null && moveString.toLowerCase().equals("save")) {
```

Comment:
as discussed, you should be able to switch this to be handled with a single function call like `getGameInput()` or something which takes in a prompt to print, grabs a line of input, and either processes global commands like `save`/`quit` or returns the input